File: backend/src/main/java/com/yourorg/appname/service/DoctorService.java
package com.yourorg.appname.service;

import com.yourorg.appname.dto.response.DoctorResponse;
import com.yourorg.appname.dto.response.DoctorReviewResponse;
import com.yourorg.appname.dto.response.ScheduleResponse;

import java.util.List;

public interface DoctorService {
    List<DoctorResponse> getAllDoctors(String search, Long departmentId, Long hospitalId);
    DoctorResponse getDoctorById(Long id);
    List<ScheduleResponse> getDoctorSchedules(Long doctorId, String date);
    List<DoctorReviewResponse> getDoctorReviews(Long doctorId);
}

File: frontend/src/services/doctorService.js
import apiClient from './apiClient';

export const doctorService = {
  async getAllDoctors({ search = '', departmentId = null, hospitalId = null } = {}) {
    const params = {};
    if (search) params.search = search;
    if (departmentId) params.departmentId = departmentId;
    if (hospitalId) params.hospitalId = hospitalId;

    const response = await apiClient.get('/doctors', { params });
    return response.data;
  },

  async getDoctorById(id) {
    const response = await apiClient.get(`/doctors/${id}`);
    return response.data;
  },

  async getDoctorSchedules(id, date = '') {
    const params = date ? { date } : {};
    const response = await apiClient.get(`/doctors/${id}/schedules`, { params });
    return response.data;
  },

  async getDoctorReviews(id) {
    const response = await apiClient.get(`/doctors/${id}/reviews`);
    return response.data;
  },
};
